iOS Client/Routing/Routing.swift

```diff
@@ -25,54 +24,28 @@ enum Route: Equatable {
     }
  
     
-    static func viewController(forRoute route: Route, inTab tab: Tab) -> UIViewController? {
+    func viewController(forTab tab: Tab) -> UIViewController? {
```

Comment:
an this is now an instance method on route component enum, so the article can be accessed from the associated value in case where self is `detail`. Thought this would be a little cleaner, but up for other opinions here . . .